scripts/build-npm.sh

```diff
@@ -35,6 +35,9 @@ for pkgJson in $(find typescript -name cdk.json | grep -v node_modules); do
         npm install
         npm run build
 
+        # cdk synth breaks in examples that rely on environment variables
+        if [[ -f DO_NOT_AUTOSYNTH ]]; then exit 0; fi
+
```

Comment:
I use a number of environment variables, and it's important for the stack to fail if any of them are not set, to avoid later errors that are harder to troubleshoot. It might be possible to replace some of them with parameter store, but this complicates setup, and I'm not sure if it's practical to get rid of all of them. I will look into that today.